src/dsymbol/tests.d

```diff
@@ -0,0 +1,68 @@
+module dsymbol.tests;
+
+import std.experimental.allocator;
+import dparse.parser, dparse.lexer, dparse.rollback_allocator;
+import dsymbol.cache_entry, dsymbol.modulecache, dsymbol.symbol;
+import std.file, std.path, std.format;
+import std.stdio : writeln, stdout;
+
+/**
+ * Parses `source`, caches its symbols and compares the the cache content
+ * with the `results`.
+ *
+ * Params:
+ *      source = The source code to test.
+ *      results = An array of string array. Each slot represents the variable name
+ *      followed by the type strings.
+ */
+version (unittest)
+void expectSymbolsAndTypes(const string source, const string[][] results)
+{
+    static immutable rName = "dsymbol-test-154251542-56564105-78944416-98523170-02452336.d";
```

Comment:
Note: Why don't you generate the id dynamically?
Also if the test suite ever gets run on multiple threads, having a hard-code file name might not be a good idea.